src/components/RenderComments/components/CommentInput/CommentInput.jsx

```diff
@@ -0,0 +1,47 @@
+import { useState, useContext } from "react"
+import { UserContext } from "../../../../contexts/UserContext"
+import { postComment } from "../../../../utils/utils"
+import { CommentItem } from "../CommentItem.jsx/CommentItem"
+import { Loading } from "../../../Loading/Loading"
+
+export const CommentInput = ({articleId}) => {
+    const [commentInputBody, setCommentInputBody] = useState('')
+    const [postedComment, setPostedComment] = useState('')
+    const [isLoading, setIsLoading] = useState(false)
+    const {user} = useContext(UserContext)
+    
+    if (!user) {
+        return <p>Login to comment</p>
+    }
+
+    const handleSubmit = (e) => {
+        e.preventDefault()
+        const sendData = {
+            "username": user,
+            "body": commentInputBody
+        }
+        //need to add some logic so the comments re-render after posting (to avoid key conflicts)
+        //tried to but it would re-render is component when I did so
+        postComment(`articles/${articleId}/comments`, sendData, setPostedComment, setIsLoading)
```

Comment:
rather than passing through the set states here, consider chaining a .then() and handling this in this file - getting into this habit is good at this stage to keep your functions flexible and re-usable